src/analyses/reaching_definitions.h

```diff
@@ -97,16 +105,28 @@ inline bool operator<(
   return false;
 }
 
-class rd_range_domaint:public ai_domain_baset
+struct infot
 {
-public:
-  rd_range_domaint():
-    ai_domain_baset(),
-    has_values(false),
-    bv_container(nullptr)
+  infot(value_setst &value_sets, is_threadedt &is_threaded, dirtyt &is_dirty)
+    : value_sets(value_sets), is_threaded(is_threaded), is_dirty(is_dirty)
+  {
+  }
+
+  value_setst &value_sets;
+  is_threadedt &is_threaded;
+  dirtyt &is_dirty;
+};
+
+template <bool remove_locals>
```

Comment:
What's the need or idea of this template parameter? Would it also be doable without templates?

Reply:
The `ait` framework creates abstract states via their default constructor. Thus a template parameter was used for configuration here instead of passing it as a flag to the contructor.